service/selectors/entities/channels.js

```diff
@@ -148,3 +149,22 @@ export const getAutocompleteChannelWithSections = createSelector(
         return channels;
     }
 );
+
+export const canManageChannelMembers = createSelector(
+    getCurrentChannel,
+    getCurrentChannelMembership,
+    getCurrentTeamMembership,
+    getUsers,
+    getCurrentUserId,
+    (channel, channelMembership, teamMembership, allUsers, currentUserId) => {
+        const user = allUsers[currentUserId];
+        const roles = `${channelMembership.roles} ${teamMembership.roles} ${user.roles}`;
```

Comment:
There's a selector available for this exact operation called```getCurrentUserRoles```. It's in ```service/selectors/entities/users```. Might be a good idea to reuse it as it would eliminate the need for ```getCurrentChannel, getCurrentChannelMembership, getCurrentTeamMembership```.